enterprise/kernel/src/test/java/org/neo4j/kernel/enterprise/builtinprocs/BuiltInProceduresProvider.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2002-2016 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.enterprise.builtinprocs;
+
+import org.neo4j.helpers.Service;
+import org.neo4j.kernel.impl.factory.ProceduresProvider;
+import org.neo4j.kernel.impl.proc.Procedures;
+
+import static org.neo4j.kernel.builtinprocs.BuiltInProceduresProvider.actuallyRegisterProcedures;
+
+@Service.Implementation( ProceduresProvider.class )
+public class BuiltInProceduresProvider extends Service implements ProceduresProvider
```

Comment:
yep i saw that different builtin procedures things but i was thinking more about why we do not have procedure registration during start up that will register correct things, but using service loaders instead?